File: benchmark/c/cpw/sum_matrix_element_element_integer_division_row_column_1/sum_matrix_element_element_integer_division_row_column_1.c


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>



int sum_matrix_element_element_integer_division_row_column_1 ( int n ) ;
int sum_matrix_element_element_integer_division_row_column_1 ( int n ) {
  int ans = 0, temp = 0, num;
  for ( int i = 1;
  i <= n && temp < n;
  i ++ ) {
    temp = i - 1;
    num = 1;
    while ( temp < n ) {
      if ( temp + i <= n ) ans += ( i * num );
      else ans += ( ( n - temp ) * num );
      temp += i;
      num ++;
    }
  }
  return ans;
}

File: experimental data/RQ3&RQ4/translation_result/success/c/cpw/mistral/BaseRepair/few-shot-True/cot-ast/candidates-3/sum_matrix_element_element_integer_division_row_column_1/sum_matrix_element_element_integer_division_row_column_1.rs
fn sum_matrix_element_element_integer_division_row_column_1(n: i32) -> i32 {
    let mut ans: i32 = 0;
    let mut temp: i32 = 0;
    let mut num: i32;
    let mut i: i32 = 1;
    while i <= n && temp < n {
        temp = i.wrapping_sub(1);
        num = 1;
        while temp < n {
            if temp.wrapping_add(i) <= n {
                ans = ans.wrapping_add(i.wrapping_mul(num));
            } else {
                ans = ans.wrapping_add((n.wrapping_sub(temp)).wrapping_mul(num));
            }
            temp = temp.wrapping_add(i);
            num = num.wrapping_add(1);
        }
        i = i.wrapping_add(1);
    }
    ans
}